ai2thor/controller.py

```diff
@@ -671,24 +737,44 @@ def step(self, action=None, **action_args):
         action = copy.deepcopy(action)
 
         # XXX should be able to get rid of this with some sort of deprecation warning
-        if 'AI2THOR_VISIBILITY_DISTANCE' in os.environ:
-            action['visibilityDistance'] = float(os.environ['AI2THOR_VISIBILITY_DISTANCE'])
+        if "AI2THOR_VISIBILITY_DISTANCE" in os.environ:
+            action["visibilityDistance"] = float(
+                os.environ["AI2THOR_VISIBILITY_DISTANCE"]
+            )
 
         self.last_action = action
 
-        rotation = action.get('rotation')
+        rotation = action.get("rotation")
         if rotation is not None and type(rotation) != dict:
-            action['rotation'] = {}
-            action['rotation']['y'] = rotation
+            action["rotation"] = {}
+            action["rotation"]["y"] = rotation
+
+        # support for deprecated parameter names (old: new)
+        changed_parameter_names = {
```

Comment:
Can you change the structure of this to the following:

changed_parameter_names[<ACTION_NAME>][<OLD_KEY>] = <NEW_KEY>

presently, we have to iterate through this list on each and every action and in theory someone could add a new mapping in there for an action and it would inadvertently remap the param name for a different action.